### PR TITLE
Add `Semigroup`/`Monoid` instances for `LspT`

### PR DESCRIPTION
**Pull request by Gabriel439**

### CODE REVIEW

**Comment by michaelpj** [comment]
A short justification would be nice :)

Do people generally like these instances? I agree they're certainly lawful, but I worry about getting worse error messages because you can now `<>` a monadic  thing when you maybe didn't mean to.

File: lsp/src/Language/LSP/Server/Core.hs [hunk 1]
```diff
@@ -64,6 +65,12 @@ import           Control.Monad.Catch (MonadMask, MonadCatch, MonadThrow)
 newtype LspT config m a = LspT { unLspT :: ReaderT (LanguageContextEnv config) m a }
   deriving (Functor, Applicative, Monad, MonadCatch, MonadIO, MonadMask, MonadThrow, MonadTrans, MonadUnliftIO, MonadFix)
 
+instance Semigroup a => Semigroup (LspT config m a) where
+    (<>) = liftA2 (<>)
+
+instance Monoid a => Monoid (LspT config m a) where
```
Comment (michaelpj) [inline lsp/src/Language/LSP/Server/Core.hs:71]: Derive these `via Ap`?

**Comment by Gabriel439** [comment]
I wrote up a post to explain why I typically add these instances: https://www.haskellforall.com/2022/03/applicatives-should-usually-implement.html

**Comment by michaelpj** [comment]
Thanks, that was a lot more explanation than I was expecting :) LGTM

**Review by michaelpj — APPROVED** [review]

**Comment by michaelpj** [comment]
Needs some language pragmas too, alas.

**Comment by michaelpj** [comment]
Thanks!

**Pull request merged**